`cv2.mat_wrapper` INTERNAL ERROR

<!--
  Use this form only if mypy reports an "INTERNAL ERROR" and/or gives a traceback.
  Please include the traceback and all other messages below (use `mypy --show-traceback`).
-->

**Crash Report**

My project fails with mypy 1.11.0, but works with 1.10.1.

It is open source, you can check [the job failing here](https://github.com/roboflow/roboflow-python/actions/runs/10013485251/job/27681191998#step:5:23).

**Traceback**

```
$ mypy . --show-traceback
roboflow-python/lib/python3.11/site-packages/cv2/mat_wrapper/__init__.py:9: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 1.11.0
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 5830, in accept
  File "mypy/nodes.py", line 2028, in accept
  File "mypy/checkexpr.py", line 4270, in visit_index_expr
  File "mypy/checkexpr.py", line 4286, in visit_index_expr_helper
  File "mypy/checkexpr.py", line 4345, in visit_index_with_type
  File "mypy/checkexpr.py", line 5863, in named_type
  File "mypy/checker.py", line 6924, in named_type
  File "mypy/checker.py", line 7048, in lookup_qualified
KeyError: "Failed qualified lookup: 'GenericAlias' (fullname = 'types.GenericAlias')."
roboflow-python/lib/python3.11/site-packages/cv2/mat_wrapper/__init__.py:9: : note: use --pdb to drop into pdb
```

**To Reproduce**

I reproduced local too, just run mypy on my project.

--- a/mypy/checkexpr.py
+++ b/mypy/checkexpr.py
@@ -4341,7 +4341,7 @@ class ExpressionChecker(ExpressionVisitor[Type]):
         elif isinstance(left_type, FunctionLike) and left_type.is_type_obj():
             if left_type.type_object().is_enum:
                 return self.visit_enum_index_expr(left_type.type_object(), e.index, e)
-            elif left_type.type_object().type_vars:
+            elif left_type.type_object().type_vars and self.chk.options.python_version >= (3, 9):
                 return self.named_type("types.GenericAlias")
             elif (
                 left_type.type_object().fullname == "builtins.type"

--- a/test-data/unit/check-functions.test
+++ b/test-data/unit/check-functions.test
@@ -1779,6 +1779,7 @@ def Arg(x, y): pass
 F = Callable[[Arg(int, 'x')], int]  # E: Invalid argument constructor "__main__.Arg"
 
 [case testCallableParsingFromExpr]
+# flags: --python-version 3.9
 from typing import Callable, List
 from mypy_extensions import Arg, VarArg, KwArg
 import mypy_extensions
@@ -1799,10 +1800,23 @@ L = Callable[[Arg(name='x', type=int)], int] # ok
 # I have commented out the following test because I don't know how to expect the "defined here" note part of the error.
 # M = Callable[[Arg(gnome='x', type=int)], int]   E: Invalid type alias: expression is not a valid type   E: Unexpected keyword argument "gnome" for "Arg"
 N = Callable[[Arg(name=None, type=int)], int] # ok
-O = Callable[[List[Arg(int)]], int]  # E: Invalid type alias: expression is not a valid type  # E: Value of type "int" is not indexable  # E: Type expected within [...]
+O = Callable[[List[Arg(int)]], int]  # E: Invalid type alias: expression is not a valid type \
+                                     # E: Value of type "int" is not indexable \
+                                     # E: Type expected within [...]
 P = Callable[[mypy_extensions.VarArg(int)], int] # ok
-Q = Callable[[Arg(int, type=int)], int]  # E: Invalid type alias: expression is not a valid type  # E: Value of type "int" is not indexable  # E: "Arg" gets multiple values for keyword argument "type"
-R = Callable[[Arg(int, 'x', name='y')], int]  # E: Invalid type alias: expression is not a valid type  # E: Value of type "int" is not indexable  # E: "Arg" gets multiple values for keyword argument "name"
+Q = Callable[[Arg(int, type=int)], int]  # E: Invalid type alias: expression is not a valid type \
+                                         # E: Value of type "int" is not indexable \
+                                         # E: "Arg" gets multiple values for keyword argument "type"
+R = Callable[[Arg(int, 'x', name='y')], int]  # E: Invalid type alias: expression is not a valid type \
+                                              # E: Value of type "int" is not indexable \
+                                              # E: "Arg" gets multiple values for keyword argument "name"
+
+
+
+
+
+
+
 [builtins fixtures/dict.pyi]
 
 [case testCallableParsing]

